src/box/xlog.c

```diff
@@ -1571,6 +1577,18 @@ xlog_cursor_ensure(struct xlog_cursor *cursor, size_t count)
 	assert((size_t)readen <= to_load);
 	ibuf_alloc(&cursor->rbuf, readen);
 	cursor->read_offset += readen;
+	/*
+	 * Grow readahead size if the requested number of bytes was successfully
+	 * read, and lower it otherwise.
+	 */
+	if ((size_t)readen == to_load) {
+		if (cursor->read_ahead * 2 <= XLOG_READ_AHEAD_MAX)
+			cursor->read_ahead *= 2;
+	} else if (cursor->read_ahead / 2 >= XLOG_READ_AHEAD_MIN) {
+		cursor->read_ahead /= 2;
```

Comment:
AFAIR we decided to set `read_ahead` to the min immediately if we read less than could. This would also be consistent with shrinking the bufffer.

Reply:
Changed to:
```c
{
        cursor->need_rbuf_shrink = true;
        cursor->read_ahead = XLOG_READ_AHEAD_MIN;
}
```